packages/react-charting/src/components/TreeChart/TreeChart.base.tsx

```diff
@@ -0,0 +1,293 @@
+
+// Create a parent class for common tree components
+class StandardTree {
+  public treeData: IDataStructure;
+  public svg: Selection<SVGSVGElement | null, unknown, null, undefined>;
+  constructor(treeData: IDataStructure, svg: Selection<SVGSVGElement | null, unknown, null, undefined>) {
+    this.treeData = treeData;
+    this.svg = svg;
+  }
+
+  public addNodeShapetoSVG(
+    name: string,
+    subname: string,
+    xCoordinate: number,
+    yCoordinate: number,
+    fillColor: string,
+    rectangleWidth: number,
+    rectangleHeight: number,
+  ) {
+    this.svg
+      .append('rect')
+      .attr('width', rectangleWidth)
+      .attr('height', rectangleHeight)
+      .attr('x', xCoordinate)
+      .attr('y', yCoordinate)
+      .attr('padding', '10px')
+      .attr('rx', '1')
+      .style('stroke', fillColor)
+      .style('stroke-width', '2px')
+      .style('fill', 'white');
+
+    this.svg
+      .append('text')
+      .style('fill', 'black')
+      .attr('dy', yCoordinate + rectangleHeight / 2.5)
+      .attr('x', () => {
+        return xCoordinate + rectangleWidth / 3.6;
+      })
+      .attr('text-anchor', () => {
+        return 'start';
+      })
+      .text(function () {
+        return name;
+      })
+      .append('tspan')
+      .attr('dy', '1.4em')
+      .attr('x', () => {
+        return xCoordinate + rectangleWidth / 3.4;
+      })
+      .text(() => {
+        return subname;
+      });
+  }
+  // Creates a rectangular path from parent to the child nodes
+  public addLinktoNodes(child: any, parent: any, leaf: boolean, rectWidth: number, gap: number): any {
+    let path = `M${child.x + rectWidth / 2},${child.y - gap} H${parent.x + rectWidth / 2} V${parent.y}`;
+    let leafpath = `M${parent.x + rectWidth / 2},${parent.y} V${parent.y + gap * 5} H${parent.x - gap / 2} H${
+      parent.x + rectWidth + gap / 2
+    }`;
+    return leaf ? leafpath : path;
+  }
+}
+// Create child class to Add Tree component based on treeHeight
+class LayeredTree extends StandardTree {
+  public composition: number | undefined;
+  constructor(
+    treeData: IDataStructure,
+    composition: number | undefined,
+    svg: Selection<SVGSVGElement | null, unknown, null, undefined>,
+  ) {
+    super(treeData, svg);
+    this.composition = composition;
+  }
+  public createTree() {
+    let root = hierarchy(this.treeData, function (d: any) {
+      return d.children;
+    });
+
+    let treeHeight = root?.height + 1;
+
+    // Create tree layout, for 2 layered width is 150, 3 layered width is 75
+    let treemap = tree().nodeSize([treeHeight === 2 ? 150 : 60, 50]);
+
+    // Assigns the x and y position for the nodes
+    let treeData = treemap(root);
+
+    // Compute the new tree layout.
+    let nodes = treeData.descendants();
+
+    // Normalize for fixed-depth and width
+    nodes.forEach(function (d) {
+      d.y = d.depth * 120 + 6;
+      d.x += 400;
```

Comment:
can you explain these numbers in comments

Reply:
// Normalise y coordinate by depth of each node by a factor of 130
// Normalise x coordinate by start coordinate 0 with 400

Added these comments.